### PR TITLE
Kindsys: Target k8s style resource definitions

### PR DESCRIPTION
**Pull request by ryantxu**
This PR updates the resource kinds so that the generated models are targeting k8s style CRD resources.  This will include:
* metadata -- must fit in k8s style metadata
* spec -- what we currently use
* status -- not yet used

Depends on https://github.com/grafana/kindsys/pull/11

In a follow up PR, we can re-introduce the devenv dashboard checks -- this will require require conversion that expects the input to target only the spec, not the whole resource.

This brings us much closer to having the kinds represent clear target shape 🎉

### CODE REVIEW

File: kinds/serviceaccount/serviceaccount_kind.cue [hunk 1]
```diff
@@ -11,38 +9,36 @@ lineage: seqs: [
...
+					avatarUrl: string @grafanamaturity(ToMetadata="kind")
+					// AccessControl metadata associated with a given resource.
+					accessControl?: {
+						[string]: bool @grafanamaturity(ToMetadata="sys")
+					}
 
-				// Teams is a list of teams the service account belongs to.
-				teams?: [...string] @grafanamaturity(ToMetadata="sys")
-				// Created indicates when the service account was created.
-				created?: string & time.Time
```
Comment (ryantxu) [inline kinds/serviceaccount/serviceaccount_kind.cue:40]: removed created+updated as they are now in standard metadata 🎉

File: go.mod [hunk 2]
```diff
@@ -423,3 +423,5 @@ replace github.com/prometheus/alertmanager => github.com/grafana/prometheus-aler
 replace google.golang.org/grpc => google.golang.org/grpc v1.45.0
 
 replace google.golang.org/genproto => google.golang.org/genproto v0.0.0-20220421151946-72621c1f0bd3
+
+replace github.com/grafana/kindsys => ../kindsys
```
Comment (ryantxu) [inline go.mod:427]: ^^ TODO:  depends on https://github.com/grafana/kindsys/pull/11

File: packages/grafana-schema/src/index.ts [hunk 3]
```diff
@@ -5,3 +5,9 @@
  */
 export * from './veneer/common.types';
 export * from './index.gen';
+
+// TODO: The following imports should be included in ./index.gen
```
Comment (ryantxu) [inline packages/grafana-schema/src/index.ts:9]: This lets our current system work as expected.  Given that we are looking to export this into a more more nested structure with `Resource<Metadata,Spec,Status>` wrappers anyway, I think this is good enough for this PR.

File: pkg/api/dashboard.go [hunk 4]
```diff
@@ -840,7 +840,10 @@ func (hs *HTTPServer) ValidateDashboard(c *contextmodel.ReqContext) response.Res
 	// work), or if schemaVersion is absent (which will happen once the Thema
 	// schema becomes canonical).
 	if err != nil || schemaVersion >= dashboard.HandoffSchemaVersion {
-		_, _, validationErr := dk.JSONValueMux(dashboardBytes)
+		// Schemas expect the dashboard to live in the spec field
+		k8sResource := `{"spec": ` + cmd.Dashboard + "}"
```
Comment (ryantxu) [inline pkg/api/dashboard.go:844]: this looks like the one place that tries to validate anything

**Review by nmarrs — COMMENTED** [review]
Don't feel confident / knowledgable enough to approve these changes, that being said they seem to make sense 👍 great job! 🎊

File: docs/sources/developers/kinds/core/dashboard/schema-reference.md [hunk 5]
```diff
@@ -13,6 +13,57 @@ title: Dashboard kind
 
 A Grafana dashboard.
 
+| Property   | Type                | Required | Default | Description                                                                                                                                                                                                                                                                   |
+|------------|---------------------|----------|---------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `metadata` | [object](#metadata) | **Yes**  |         | metadata contains embedded CommonMetadata and can be extended with custom string fields<br/>TODO: use CommonMetadata instead of redfining here; currently needs to be defined here<br/>without external reference as using the CommonMetadata reference breaks thema codegen. |
```
[the remark this answers is not in the document]
Comment (sdboyer) [reply docs/sources/developers/kinds/core/dashboard/schema-reference.md:18]: these docs are all generated from comments in the original `dashboard_kind.cue` file - the update needs to be made there

File: .github/CODEOWNERS [hunk 6]
```diff
@@ -473,7 +473,6 @@ lerna.json @grafana/frontend-ops
 /scripts/trigger_docker_build.sh @grafana/grafana-delivery
 /scripts/trigger_grafana_packer.sh @grafana/grafana-delivery
 /scripts/trigger_windows_build.sh @grafana/grafana-delivery
-/scripts/validate-devenv-dashboards.sh @grafana/grafana-delivery
```
Comment (ryantxu) [inline .github/CODEOWNERS:476]: this was an unused/broken script

File: pkg/api/dashboard.go (hunk 4 above)
Comment (sdboyer) [reply pkg/api/dashboard.go:844]: yep, this should be the only one, since we don't have the devenv validation anymore

**Review by sdboyer — APPROVED** [review]
i fixed the veneer index generator so we don't need the manual hack. 🎉 

There's like half a dozen follow-ups to be done here in order to clean up some messes we've left behind, but i don't think we should treat those as blockers.

**Pull request merged**